source/exe/signal_action.cc

```diff
@@ -20,6 +46,14 @@ void SignalAction::sigHandler(int sig, siginfo_t* info, void* context) {
   }
   tracer.logTrace();
 
+  FailureFunctionList* list = crash_handlers.exchange(nullptr, std::memory_order_relaxed);
+  if (list) {
+    // Finally after logging the stack trace, call any registered crash handlers.
+    for (const auto* handler : *list) {
+      handler->crashHandler();
```

Comment:
FWIW, I think the term "crash" here is not precise enough. For example, these signals may be generated by a `kill` command, which is not a crash in the usual sense. I think if we use "fatal" instead, as is done elsewhere in this file, the intuition is a bit clearer.